server-common/src/test/java/org/apache/kafka/server/immutable/pcollections/PCollectionsImmutableMapTest.java

```diff
@@ -225,77 +225,59 @@ public void testDelegationOfReplaceAll() {
         new PCollectionsHashMapWrapperDelegationChecker<>()
             .defineMockConfigurationForVoidMethodInvocation(mock -> mock.replaceAll(eq(mockBiFunction)))
             .defineWrapperVoidMethodInvocation(wrapper -> wrapper.replaceAll(mockBiFunction))
-            .doVoidMethodDelegationCheck();
+            .doUnsupportedVoidFunctionDelegrationCheck();
     }
 
     @Test
     public void testDelegationOfPutIfAbsent() {
         new PCollectionsHashMapWrapperDelegationChecker<>()
-            .defineMockConfigurationForFunctionInvocation(mock -> mock.putIfAbsent(eq(this), eq(this)), this)
-            .defineWrapperFunctionInvocationAndMockReturnValueTransformation(wrapper -> wrapper.putIfAbsent(this, this), identity())
-            .doFunctionDelegationCheck();
-    }
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    public void testDelegationOfRemoveByKeyAndValue(boolean mockFunctionReturnValue) {
-        new PCollectionsHashMapWrapperDelegationChecker<>()
-            .defineMockConfigurationForFunctionInvocation(mock -> mock.remove(eq(this), eq(this)), mockFunctionReturnValue)
-            .defineWrapperFunctionInvocationAndMockReturnValueTransformation(wrapper -> wrapper.remove(this, this), identity())
-            .doFunctionDelegationCheck();
-    }
-
-    @ParameterizedTest
-    @ValueSource(booleans = {true, false})
-    public void testDelegationOfReplaceWhenMappedToSpecificValue(boolean mockFunctionReturnValue) {
-        new PCollectionsHashMapWrapperDelegationChecker<>()
-            .defineMockConfigurationForFunctionInvocation(mock -> mock.replace(eq(this), eq(this), eq(this)), mockFunctionReturnValue)
-            .defineWrapperFunctionInvocationAndMockReturnValueTransformation(wrapper -> wrapper.replace(this, this, this), identity())
-            .doFunctionDelegationCheck();
```

Comment:
Looks like we lost these two test methods

Reply:
These two functions are not overridden by PCollection classes. `mock.replace()` or `mock.remove()` will call Java's default implementation in `Map.replace()` or `AbstractMap.remove()`. Mocking that testing that seemed not required.

For example: `Map.replace()` looks like this: 
```
default boolean replace(K key, V oldValue, V newValue) {
        Object curValue = get(key);
        if (!Objects.equals(curValue, oldValue) ||
            (curValue == null && !containsKey(key))) {
            return false;
        }
        put(key, newValue);
        return true;
    }
```
Since it internally calls put which we are already testing, so that's why I thought it was not necessary to test these functions. Likewise for `remove()`.

Reply:
Yeah, this is why I said in the comment above:
```
I think on the one hand we don't have to test that the real method throws the exception -- that's up to the library itself to test in its test suite. All we have to test is that we delegate correctly. 
```

The underlying library implementation in I guess these two cases doesn't explicitly throw the exception but instead reuses a default implementation from `java.util`, and that will cause an exception to be thrown if the map ends up needing to be mutated.

At a minimum we should put the two test cases back in as I wrote them.  If you want to do it with an explicit check for the thrown exception by invoking the underlying method then you may have to do a bit more configuration on the mock so that the underlying method will actually decide it needs to mutate the map.  probably just putting my stuff back in is easier (and also add "UnsupportedOperation" into the test method names).